diff --git a/src/dvdnav.c b/src/dvdnav.c
index 39d83b0..2afcf2d 100644
--- a/src/dvdnav.c
+++ b/src/dvdnav.c
@@ -29,7 +29,9 @@
 #include <inttypes.h>
 #include <stdlib.h>
 #include <stdio.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <limits.h>
 #include <string.h>
 #include <sys/time.h>
diff --git a/src/dvdnav_internal.h b/src/dvdnav_internal.h
index 495f032..e2052a6 100644
--- a/src/dvdnav_internal.h
+++ b/src/dvdnav_internal.h
@@ -232,7 +232,7 @@ int64_t dvdnav_convert_time(dvd_time_t *time);
 #define printerrf(format, args...) \
     do { if (this) snprintf(this->err_str, MAX_ERR_LEN, format, ## args); } while (0)
 #else
-#ifdef _MSC_VER
+#if defined(_MSC_VER) && ( _MSC_VER < 1800 )
 #define printerrf(str) \
     do { if (this) snprintf(this->err_str, MAX_ERR_LEN, str); } while (0)
 #else
diff --git a/src/read_cache.c b/src/read_cache.c
index 1b2862d..e12bd6d 100644
--- a/src/read_cache.c
+++ b/src/read_cache.c
@@ -87,7 +87,7 @@ struct read_cache_s {
 # if READ_CACHE_TRACE
 #  define dprintf(fmt, ...) fprintf(MSG_OUT, "libdvdnav: %s: "fmt,  __func__ , __VA_ARGS__)
 # else
-#ifdef _MSC_VER
+#if defined(_MSC_VER) && (_MSC_VER < 1800)
 #  define dprintf(fmt, str) /* Nowt */
 #else
 #  define dprintf(fmt, ...) /* Nowt */
diff --git a/src/vm/vm.c b/src/vm/vm.c
index 6c64e59..480ff5a 100644
--- a/src/vm/vm.c
+++ b/src/vm/vm.c
@@ -28,7 +28,9 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <inttypes.h>
 #include <limits.h>
 #include <assert.h>
